core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.ts

```diff
@@ -0,0 +1,138 @@
+import { Subject } from 'rxjs';
+
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    OnDestroy,
+    OnInit
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { AutoCompleteModule } from 'primeng/autocomplete';
+import { ButtonModule } from 'primeng/button';
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { DotAddStyleClassesDialogStore } from './store/add-style-classes-dialog.store';
+
+import { StyleClassModel } from '../../models/models';
+
+const COMMA_SPACES_REGEX = /(,|\s)(.*)/;
+
+@Component({
+    selector: 'dotcms-add-style-classes-dialog',
+    standalone: true,
+    imports: [AutoCompleteModule, FormsModule, ButtonModule],
+    providers: [DotAddStyleClassesDialogStore],
+    templateUrl: './add-style-classes-dialog.component.html',
+    styleUrls: ['./add-style-classes-dialog.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AddStyleClassesDialogComponent implements OnInit, AfterViewInit, OnDestroy {
+    classes: StyleClassModel[];
+
+    selectedClasses: StyleClassModel[] = [];
+
+    filteredClasses: StyleClassModel[];
+
+    private autoCompleteInput: HTMLInputElement;
+
+    private destroy$: Subject<void> = new Subject<void>();
```

Comment:
All this looks like a lot of state... do we need to move this to the store?

Reply:
Will work on a refactor to handle this from the store